genie-core/src/main/java/com/netflix/genie/core/jpa/entities/JobEntity.java

```diff
@@ -140,9 +256,73 @@
             @JoinColumn(name = "application_id", referencedColumnName = "id", nullable = false)
         }
     )
-    @OrderColumn(name = "application_order", nullable = false)
+    @OrderColumn(name = "application_order", nullable = false, updatable = false)
     private List<ApplicationEntity> applications = new ArrayList<>();
 
+    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
```

Comment:
Because there is no repository for the `Criterion` entities we're only creating and deleting them as part of a given job. The cascade enables the JPA manager to delete all the associated entities. Though our database ddl's do this as well it is good to have this in case some entity state is stuck in the Entity Manager we want to make sure it's flushed out. Better safe than sorry.